engine/src/QuadTree.cpp

```diff
@@ -0,0 +1,233 @@
+//
+// Created by tdancois on 7/31/16.
+//
+
+#include <SFML/Graphics/Text.hpp>
+#include <sstream>
+#include "QuadTree.h"
+
+QuadTree::QuadTree(const unsigned int &x,
+                   const unsigned int &y,
+                   const unsigned int &width,
+                   const unsigned int &height,
+                   const unsigned int &node_capacity_,
+                   const unsigned int &level_) :
+        node_capacity(node_capacity_),
+        level(level_),
+        northWest(nullptr),
+        northEast(nullptr),
+        southWest(nullptr),
+        southEast(nullptr),
+        isSplit(false) {
+
+    shape.setPosition(x, y);
+
+    sf::Vector2f size(width, height);
+    shape.setSize(size);
+}
+
+QuadTree::~QuadTree() {
+    delete northEast;
```

Comment:
You should use `std::unique_ptr` instead and make the quadtree non-copyable. That would prevent a lot of future problem. Moreover, we will often use this structure to optimize our collision detection check. We cannot take the risk to have memory problem when they can be avoided.

( Have a look here if you're a bit lost : http://www.umich.edu/~eecs381/handouts/C++11_smart_ptrs.pdf )

If you don't know how to use smart pointers let me know and I'll fixup your PR.